test/system_SUITE.erl

```diff
@@ -0,0 +1,446 @@
+-module(system_SUITE).
+-compile([export_all]).
+
+-include_lib("common_test/include/ct.hrl").
+-include_lib("amqp_client/include/amqp_client.hrl").
+
+-define(SERVER_REJECT_CLIENT, {tls_alert,"unknown ca"}).
+
+all() ->
+    [
+     library,
+     invasive_SSL_option_change,
+     validation_success_for_AMQP_client,
+     validation_failure_for_AMQP_client,
+     whitelisted_certificate_accepted_from_AMQP_client_regardless_of_validation_to_root,
+     removed_certificate_denied_from_AMQP_client,
+     installed_certificate_accepted_from_AMQP_client,
+     whitelist_directory_DELTA,
+     replaced_whitelisted_certificate_should_be_accepted,
+     ensure_configuration_using_binary_strings_is_handled
+    ].
+
+suite() ->
+    [{timetrap, {seconds, 25}}].
+
+%% -------------------------------------------------------------------
+%% Testsuite setup/teardown.
+%% -------------------------------------------------------------------
+
+init_per_suite(Config) ->
+    rabbit_ct_helpers:log_environment(),
+    Config1 = rabbit_ct_helpers:set_config(Config, [
+        {rmq_nodename_suffix, ?MODULE},
+        {rmq_extra_tcp_ports, [tcp_port_amqp_tls_extra]}
+      ]),
+    rabbit_ct_helpers:run_setup_steps(Config1,
+      rabbit_ct_broker_helpers:setup_steps() ++
+      rabbit_ct_client_helpers:setup_steps()).
+
+end_per_suite(Config) ->
+    rabbit_ct_helpers:run_teardown_steps(Config,
+      rabbit_ct_client_helpers:teardown_steps() ++
+      rabbit_ct_broker_helpers:teardown_steps()).
+
+init_per_group(_, Config) ->
+    Config.
+
+end_per_group(_, Config) ->
+    Config.
+
+init_per_testcase(Testcase, Config) ->
+    TestCaseDir = rabbit_ct_helpers:config_to_testcase_name(Config, Testcase),
+    WhitelistDir = filename:join([?config(priv_dir, Config), TestCaseDir]),
```

Comment:
There are already quite a lot of data written to `priv_dir` by `rabbit_ct_helpers` and `rabbit_ct_broker_helpers`, including the "normal" TLS certificates.

To avoid conflict with other data, I suggest you use either a namespace (a subdirectory) specific to your testsuite or you put your certificates in a subdirectory of the existing certs dir, eg.:

```erlang
WhitelistDir = filename:join([?config(rmq_certsdir, Config), "extra", TestCaseDir]),
```

`rmq_certsdir` is a subdirectory in `priv_dir`. Sorry, I should have mentionned it already...